File: ruby/algo_ladder/reverse_words/reverse_words_explained.rb
# Given a string of words, return a new string that contains the words in reverse order.

# Input: “popcorn is so cool isn’t it yeah i thought so”
# Output: “so thought i yeah it isn’t cool so is popcorn”

# Create function that takes in a string
def reverse_words (input_string)
  # Split the string whereever there are spaces
  split_string_as_array = input_string.split(" ")
  # Reverse the array
  reversed_array_of_strings = split_string_as_array.reverse
  # Join the array wherever there are spaces
  reversed_string = reversed_array_of_strings.join(" ")
  # Return the string that is now reversed
  return reversed_string
end
# Call the function with its text input
p reverse_words("popcorn is so cool isn’t it yeah i thought so")

File: javascript/algoLadder/reverseWords/reverseWords_explained.js
// Given a string of words, return a new string that contains the words in reverse order.

// Input: “popcorn is so cool isn’t it yeah i thought so”
// Output: “so thought i yeah it isn’t cool so is popcorn”

// Create function that takes in a string
const reverseWords = function (string) {
  // Split string and create new variable
  let splitString = string.split(" ")
  // Reverse the split string and create it as a new variable
  let reversedSplitString = splitString.reverse()
  // Join the split string back together and save it as a new variable
  let joinedReversedString = reversedSplitString.join(" ")
  // Return the split string
  return joinedReversedString
}

// Call the function with the test input string
console.log(reverseWords("popcorn is so cool isn’t it yeah i thought so"))